tests/src/Traits/TestDocumentTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\Traits;
+
+use Drupal\media\MediaInterface;
+use Drupal\Tests\RandomGeneratorTrait;
+
+/**
+ * Contains a method to create a test document.
+ */
+trait TestDocumentTrait {
+
+  use RandomGeneratorTrait;
+  use MediaCreationTrait;
+
+  /**
+   * A media entity to test with.
+   */
+  protected MediaInterface $media;
+
+  /**
+   * Enables a standalone media page.
+   *
+   * This must be called before ->visitMediaPage().
+   */
+  protected function enableStandaloneMediaPage(): void {
+    \Drupal::configFactory()
+      ->getEditable('media.settings')
+      ->set('standalone_url', TRUE)
+      ->save(TRUE);
+
+    $this->container->get('router.builder')->rebuild();
+  }
+
+  /**
+   * Creates a test document, and sets it in a property.
+   *
+   * This must be called before ->visitMediaPage().
+   */
+  protected function createTestDocument(): void {
+    $this->media = $this->createMediaEntity('document', [
```

Comment:
I don't see much value in this, and I don't like objects in properties of classes. Let's drop it.

Reply:
I dropped the trait and the property.\
I also updated the modal formatter branch with this change.